### PR TITLE
Create PaymentMethods.h

### PR DESCRIPTION
**Pull request by AhmedHamdyKotp**

### CODE REVIEW

File: include/PaymentMethods.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// Created by Ahmed Hamdy Kotp
+
+#ifndef PAYMENTMETHODS_H
+#define PAYMENTMETHODS_H
+
+#include <string>
+#include <vector>
+#include "Date.h" 
+
+class PaymentMethods {
```
Comment (OsamaAshraf01) [inline include/PaymentMethods.h:11]: PaymentMethod*

File: include/PaymentMethods.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+private:
+    vector<Card> cards;
+    vector<DigitalWallet> wallets;
+    vector<BankTransfer> transfers;
```
Comment (OsamaAshraf01) [inline include/PaymentMethods.h:16]: Separate each one in an independent class.
class Card, DigitalWallet, BankTransfer. All of them will inherit from PaymentMethod Class.
---
The user will write something like:
```C++
Card c{"1234-5678-9012-3456", 111, {1, 1, 2026}, 123};
c.showDetails();

// In User Class
vector<Card> cards;
vector<DigitalWallet> wallets;
vector<BankTransfer> transfers;
// User will have an array of each type of Payment Methods
```

File: include/PaymentMethods.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+
+    //Attributes
+    struct Card {
+        string card_num;
+        int cvv_num;
+        Date expiry_date; 
+        double Money;
+    };
```
Comment (OsamaAshraf01) [inline include/PaymentMethods.h:30]: This will be moved to Card Class and so on.

File: include/PaymentMethods.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    struct DigitalWallet {
+        int id;
+        string address;
+        string currencyType;
+        double Money;
```
Comment (OsamaAshraf01) [inline include/PaymentMethods.h:36]: Can be replaced by Currency Class in the future.

File: include/PaymentMethods.h (hunk 2 above)
Comment (OsamaAshraf01) [reply include/PaymentMethods.h:16]: The PaymentMethod then will contain the general attributes and methods between all payment methods, such as currency type, transaction fees, time constraints between transactions, and security factors.

**Review by OsamaAshraf01 — CHANGES_REQUESTED** [review]
Very excited to see the results !!

File: include/PaymentMethods.h (hunk 2 above)
Comment (AhmedHamdyKotp) [reply include/PaymentMethods.h:16]: Yes, this idea looks pretty good,  but my back will be brocken unfortunately 😭

Comment (OsamaAshraf01) [reply include/PaymentMethods.h:16]: Why my friend? it's just some constructors, getters, setters, and methods 😅😅. Don't get tired now, we still have a lot of work !!🔥🔥

Comment (OsamaAshraf01) [reply include/PaymentMethods.h:16]: You already made almost 75% of the work.

**Pull request closed without merging**